src/main/host/descriptor/socket/netlink.rs

```diff
@@ -689,8 +689,28 @@ impl InitialState {
             let nlmsg_type = u16::from_ne_bytes(nlmsg_type.try_into().unwrap());
 
             match nlmsg_type {
-                RTM_GETLINK => self.handle_ifinfomsg(common, &packet_buffer[..]),
-                RTM_GETADDR => self.handle_ifaddrmsg(common, &packet_buffer[..]),
+                RTM_GETLINK => {
+                    let min_len =
+                        std::mem::size_of::<nlmsghdr>() + std::mem::size_of::<ifinfomsg>();
+                    // Pad the message if it's too short and update the len field
+                    if packet_buffer.len() < min_len {
+                        packet_buffer.resize(min_len, 0);
+                        packet_buffer[memoffset::span_of!(nlmsghdr, nlmsg_len)]
+                            .copy_from_slice(&(min_len as u32).to_le_bytes()[..]);
+                    }
```

Comment:
Can you add a `debug!()` call here to say that we're zero-filling the buffer?

And can you also add a comment saying that we typically try not to zero-fill structs when the bytes are missing, but it should be okay here since we don't yet support most of the fields of `ifaddrmsg`?